pkg/clusteragent/autoscaling/cluster/config_retriever_values.go

```diff
@@ -0,0 +1,103 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package cluster
+
+import (
+	"encoding/json"
+
+	kubeAutoscaling "github.com/DataDog/agent-payload/v5/autoscaling/kubernetes"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type autoscalingValuesProcessor struct {
+	store *store
+
+	processed           map[string]struct{}
+	lastProcessingError bool
+}
+
+func newAutoscalingValuesProcessor(store *store) autoscalingValuesProcessor {
+	return autoscalingValuesProcessor{
+		store: store,
+	}
+}
+
+func (avp *autoscalingValuesProcessor) preProcess() {
```

Comment:
Given the way the reconcile works (i.e. follower does absolutely nothing), you can simplify that and just have a straightforward process.